packages/datadog-plugin-mocha/src/index.js

```diff
@@ -100,16 +100,33 @@ function createWrapRunTest (tracer, testEnvironmentMetadata, sourceRoot) {
   }
 }
 
+function getAllTestsInSuite (root) {
+  const tests = []
+  function getTests (suiteOrTest) {
+    suiteOrTest.tests.forEach(test => {
+      tests.push(test)
+    })
+    suiteOrTest.suites.forEach(suite => {
+      getTests(suite)
+    })
+  }
+  getTests(root)
+  return tests
+}
+
 // Necessary to get the skipped tests, that do not go through runTest
 function createWrapRunTests (tracer, testEnvironmentMetadata, sourceRoot) {
   return function wrapRunTests (runTests) {
     return function runTestsWithTrace () {
       runTests.apply(this, arguments)
-      this.suite.tests.forEach(test => {
+      const suite = arguments[0]
+      const tests = getAllTestsInSuite(suite)
+      tests.forEach(test => {
         const { pending: isSkipped } = test
-        if (!isSkipped) {
+        if (!isSkipped || test.__datadog_skipped) {
```

Comment:
necessary, because this function is called multiple times with different suites, and since we call `getAllTestsInSuite` with the root, all the skipped tests should already be skipped

Reply:
Can you add this comment as a code comment?

Reply:
sure 👍